eras/shelley/impl/src/Cardano/Ledger/Shelley/Rewards.hs

```diff
@@ -220,14 +220,12 @@ instance CC.Crypto c => FromCBOR (PoolRewardInfo c) where
       )
 
 notPoolOwner ::
-  HasField "_protocolVersion" pp ProtVer =>
-  pp ->
   PoolParams c ->
   Credential 'Staking c ->
   Bool
-notPoolOwner pp pps = \case
+notPoolOwner pps = \case
   KeyHashObj hk -> hk `Set.notMember` ppOwners pps
-  ScriptHashObj _ -> HardForks.allowScriptStakeCredsToEarnRewards pp
+  _ -> True
```

Comment:
It is often good to exhaustively list all patterns. Whenever the sum type is modified we get warnings about non-exhaustive pattern matches in all such places. This is not terribly important, since I don't think we plan on changing `Credential` type, but just in case it is a good practice in general
```suggestion
  ScriptHashObj _ -> True
```